### DISCUSSION TITLE
Why do commands like `task modify 3 due:today ` and `task done 3` still work as expected and where is this behavior documented?

### DISCUSSION THREAD

**Discussion by sdondley** [GothenburgBitFactory/taskwarrior#2759] [Q&A]
So when I was newer to TW and had less of a clue. I wrote a bunch of aliases and functions in the shell to make input easier. I'm now rewriting my little library to capitalize on new stuff I've learned since I started with TW a couple of years back.

Looking at my old scripts, I'm noticing some commands are wrong, although they seem to have always worked. Example:

`alias td=task done "$@"`

`alias tm=task modify "$@"`

But according to the documentation, the the task ids should go before the `done` command and any modifications should go after the `modify` command. It seems the developers decided to modify the parser to accommodate bad user input and turn it into valid input, AFAICT.

I'm wondering if this is documented somewhere and what exactly the rules are so I might take advantage of this little quirk and make processing commands easier when I rewrite my library. Thanks!

(and yes, I'm now aware you can put aliases in the `rc` file and do stuff like `t 23 m description`. Though it is still nice not to have to type the space).

**Comment by jaker-dotcom** [comment]
I think I've read somewhere that where unambiguous the commands work nonetheless. This is especially the case for commands that take filters only.
Anyways this shouldn't open up new possibilities so I'd recommend to stick to the docs. For fancier aliases look at shell expansions maybe. 

This for example places the first parameter that follows tA in the filter and all following params into the modifications space:
```
taskannotatefunction ()
    task $1 annotate -- ${@:2}
}
alias tA=taskannotatefunction
```

Or this one that sets the dependecy of given task range (e.g. 1-5,8 without spaces) to all following tasks (after a space):
```
taskmodifydependencyfunction () {
	echo "make task $1 depend on $(echo "$(task ${@:2} ids)")"
	task $1 modify depends:$(echo "$task ${@:2} ids)" | tr ' ' ',')
}
alias tad="taskmodifydependencyfunction"
``` 
Note that only the first parameter can be a task range since depends: does not work with ranges.

Hope I could help and spark a little inspiration. Feel free to ask for help although I myself am just getting into all this.
